projects/extension/sql/idempotent/908-text-to-sql-openai.sql

```diff
@@ -365,7 +369,11 @@ begin
           , jsonb_build_object('role', 'user', 'content', _prompt)
           )
         , tools=>_tools
-        , tool_choice=>'required'
+        , tool_choice=>
+            case when _iter_remaining = 0 then
+            '{"type": "function", "function": {"name": "answer_user_question_with_sql_statement"}}'
+            else 'required'
+            end
```

Comment:
Why not do this with at least anthropic as well?

Reply:
We should.